components/InfoPane/FileUploadControl.tsx

```diff
@@ -52,23 +61,18 @@ export default function FileUploadControl(props: HTMLProps<HTMLLabelElement>) {
     readFile(file);
   }
 
-  async function readFile(file: File) {
-    const reader = new FileReader();
-
-    const content: string = await new Promise(resolve => {
-      reader.onload = () => resolve(reader.result as string);
-      reader.readAsText(file);
-    });
-
+  function loadPackageJson(json: string, filename?: string): void {
     // Parse module and insert into cache
     let pkg: PackageJSON;
     try {
-      pkg = JSON.parse(content);
+      pkg = JSON.parse(json);
     } catch {
-      flash(`${file.name} is not a valid JSON file`);
+      flash(`${filename ?? 'Pasted content'} is not a valid JSON file`);
       return;
     }
 
+    flash(`Loading ${filename ?? 'pasted content'}...`);
```

Comment:
I dropped this line. I was looking for a way to give confirmation that the tool is _doing something_ but it's sometimes confusing, because this notification lasts longer than it takes to actually parse the file (which is 0ms in case of repeated pastes)

We might have to think of better ways to give feedback, for example by implementing mandatory delays—that is, _always_ show a minimum "loading" state so that the user knows something is happening, like via https://github.com/smeijer/spin-delay